pkg/controller/virtualmachine/virtualmachine_controller.go

```diff
@@ -108,19 +112,19 @@ func (r *ReconcilePolicy) Reconcile(request reconcile.Request) (reconcile.Result
 		// The object is not being deleted, so if it does not have a finalizer,
 		// then lets add the finalizer and update the object.
 		if !helper.ContainsString(instance.ObjectMeta.Finalizers, pool_manager.RuntimeObjectFinalizerName) {
-			err = r.addFinalizerAndUpdate(&request)
+			err = r.addFinalizerAndUpdate(&request, logger)
 		}
 
 		return reconcile.Result{}, err
 	}
 	// The object is being deleted
 	logger.V(1).Info("The VM is being marked for deletion")
-	err = r.removeFinalizerAndReleaseMac(instance, &request)
+	err = r.removeFinalizerAndReleaseMac(instance, &request, logger)
 	return reconcile.Result{}, err
 }
 
-func (r *ReconcilePolicy) addFinalizerAndUpdate(request *reconcile.Request) error {
-	logger := log.WithName("addFinalizerAndUpdate").WithValues("virtualMachineName", request.Name, "virtualMachineNamespace", request.Namespace)
+func (r *ReconcilePolicy) addFinalizerAndUpdate(request *reconcile.Request, reconcileLog logr.Logger) error {
```

Comment:
It does not matter to this method that the logger is from reconciler. For us it is just a logger. Maybe `parentLogger`?